### PR TITLE
Add a GraphQL API

### LINKED ISSUE #89: GraphQL support
Github's GraphQL "v4" API seems to have some attractive properties, like being able to do complex custom queries in a single request. Also, Github support just told me that there's functionality that's *only* accessible through the GraphQL API. Specifically I asked how you use the API to add a reaction to a pull request review, which is [mysteriously missing from their docs](https://developer.github.com/v3/reactions/), and they told me:

> 
> Thanks for following up, Nathaniel. This is not possible via the REST API, but is possible via the GraphQL API using this mutation:
> 
> https://developer.github.com/v4/mutation/addreaction/
> 
> Notice the supported subject types:
> 
> https://developer.github.com/v4/interface/reactable/
> 
> Hope this helps.

This isn't super critical functionality or anything, but it does seem like we might want the option to use GraphQL more.

Has anyone already figured out how to use GraphQL in a gidgethub-based framework? I would love to see some examples, b/c I find the Github docs largely impenetrable :-).

Are there new features that gidgethub should grow to support GraphQL better? I have no idea but maybe someone does.

CC: @wgwz @webknjaz

### PR DESCRIPTION
**Pull request by brettcannon**
Closes #89

### CODE REVIEW

File: gidgethub/abc.py [hunk 1]
```diff
@@ -198,3 +208,57 @@ def __init__(
...
+            raise BadGraphQLRequest(http.HTTPStatus(status_code), response)
+        elif status_code == 200:
+            self.rate_limit = sansio.RateLimit.from_http(response_headers)
+            if "errors" in response:
+                raise QueryError(response)
+            if "data" in response:
+                return response["data"]
+            else:
+                raise GraphQLException(
+                    f"response did not contain 'errors' or 'data': {response}", response
```
Comment (Mariatta) [inline gidgethub/abc.py]: (Non-blocking) 
```suggestion
                    f"Response did not contain 'errors' or 'data': {response}", response
```

File: gidgethub/abc.py [hunk 2]
```diff
@@ -198,3 +208,57 @@ def __init__(
...
+                )
+        else:
+            raise GraphQLException(
+                f"unexpected HTTP response to GraphQL request: {status_code}", response
```
Comment (Mariatta) [inline gidgethub/abc.py]:
```suggestion
                f"Unexpected HTTP response to GraphQL request: {status_code}", response
```

File: docs/abc.rst [hunk 3]
```diff
@@ -250,3 +250,19 @@ experimental APIs without issue.
         .. versionchanged:: 3.0
 
             Added *jwt* and *oauth_token*.
+
+    .. coroutine:: graphql(query, *, endpoint, **variables)
```
Comment (webknjaz) [inline docs/abc.rst]: Looks like this is missing the default `endpoint="https://api.github.com/graphql"`

Comment (brettcannon) [reply docs/abc.rst]: It's now fixed. Thanks for the catch (which happens to be for a feature I included specifically for you 😄 )!

Comment (webknjaz) [reply docs/abc.rst]: Thanks! :laughing:

**Pull request merged**